### PR TITLE
Add reset_cache function to delete the __cache__ directory

### PR DESCRIPTION
**Pull request by JuanBindez**
### Notes

- Added the `reset_cache` function to `helpers.py` to delete the cache directory (`__cache__`).
- The function checks if the directory exists and removes it if so.
- This helps in clearing cached data when needed.

#### When you think you should reset the cache, when using the use_oauth parameter, just import the function from helpers and call reset_cache()
```python
from pytubefix import YouTube
from pytubefix.cli import on_progress]
from pytubefix.helpers import reset_cache

reset_cache()
 
url = "url"

yt = YouTube(url, use_oauth=True, allow_oauth_cache=True, on_progress_callback = on_progress)
ys = yt.streams.get_highest_resolution()

ys.download()

```

### CODE REVIEW

**Comment by felipeucelli** [comment]
Good idea, but what do you think about adding `reset_cache` to innertube to check `allow_cache` and reset it if it is `False`?

We could also replace `print` with `logger.debug`, I believe that would be a good practice.

**Comment by JuanBindez** [comment]
> Good idea, but what do you think about adding `reset_cache` to innertube to check `allow_cache` and reset it if it is `False`?
> 
> We could also replace `print` with `logger.debug`, I believe that would be a good practice.

You're right, I'm thinking about keeping it this way and being able to call the reset_cache function, so if the user can call reset at a certain time without having to change the allow_cache, having both I think would be interesting.


I'll do it like this, I'll add reset_cache to allow_cache=False and also leave the option to call it

File: pytubefix/helpers.py [hunk 1]
```diff
@@ -333,7 +334,45 @@ def create_mock_html_json(vid_id) -> Dict[str, Any]:
...
+
+    Raises:
+        None
+    """
+    
+    cache_dir = os.path.join(os.path.dirname(__file__), '__cache__')
+
+    if verbose:
+        setup_logger(level=logging.DEBUG)
+
```
Comment (felipeucelli) [inline pytubefix/helpers.py:373]: I think that specifying verbose is not necessary, we can get all the logs using for example: 

```python
logging.basicConfig(level=logging.DEBUG, format='%(asctime)s - %(name)s - %(levelname)s - %(message)s')
```

Comment (JuanBindez) [reply pytubefix/helpers.py:373]: I'll keep it like this, by default it won't show anything

**Pull request merged**